src/IECoreScene/MeshPrimitive.cpp

```diff
@@ -195,6 +226,122 @@ PolygonIterator MeshPrimitive::faceEnd()
 	return PolygonIterator( m_verticesPerFace->readable().end(), m_vertexIds->readable().end(), m_vertexIds->readable().size() );
 }
 
+void MeshPrimitive::setCorners( const IECore::IntVectorData *ids, const IECore::FloatVectorData *sharpnesses )
+{
+	for( auto id : ids->readable() )
+	{
+		if( id < 0 || (size_t)id >= m_numVertices )
+		{
+			throw Exception( boost::str(
+				boost::format(
+					"Bad corners : id (%1%) is out of expected range (0-%2%)"
+				) % id % (m_numVertices - 1)
+			) );
+		}
+	}
+
+	if( sharpnesses->readable().size() != ids->readable().size() )
+	{
+		throw Exception( boost::str(
+			boost::format(
+				"Bad corners : number of sharpnesses (%1%) does not match number of ids (%2%)"
+			) % sharpnesses->readable().size() % ids->readable().size()
+		) );
+	}
+
+	m_cornerIds = ids->copy();
+	m_cornerSharpnesses = sharpnesses->copy();
+}
+
+const IECore::IntVectorData *MeshPrimitive::cornerIds() const
+{
+	return m_cornerIds ? m_cornerIds.get() : emptyIntVectorData();
+}
+
+const IECore::FloatVectorData *MeshPrimitive::cornerSharpnesses() const
+{
+	return m_cornerSharpnesses ? m_cornerSharpnesses.get() : emptyFloatVectorData();
+}
+
+void MeshPrimitive::removeCorners()
+{
+	m_cornerIds = emptyIntVectorData();
+	m_cornerSharpnesses = emptyFloatVectorData();
+}
+
+void MeshPrimitive::setCreases( const IECore::IntVectorData *lengths, const IECore::IntVectorData *ids, const IECore::FloatVectorData *sharpnesses )
+{
+	size_t expectedIds = 0;
+	for( auto length : lengths->readable() )
+	{
+		if( length < 2 || (size_t)length > m_numVertices )
```

Comment:
Just for me to understand - is the following invalid?

```
My mesh (m_numVertices = 6):

4 ---- 5
|      |
2 ---- 3
|      |
0 ---- 1

setCreases( [7], [3, 1, 0, 2, 4, 5, 3, 2], 1.0 )
```

That might be a little weird, but should technically be legal or am I missing something? :) Will most likely not come up in practice...

Reply:
You're right - I've removed that constraint in 43e5385.